### ISSUE TITLE
Reusable Header Component

### ISSUE THREAD

**Issue by D3Portillo** [Arena-Mayhem/frontend#13]
## TL;DR

Create a reusable component to be used in each view as a title.

## Component Story

- As a user I want to see a title text that can be shortened for mobile views
- As a user I want to have an optional quick action to click/touch
- As a user I want to see an image/icon left to the title
- I want to see the title + actions in a card with shadows + border
- The text must shrink when in mobile view so it fits the screen

### Component API

```ts
Header({ 
 endEnhancer?: JSX.Element, // Optional
 desktopTitle: string
 mobileTitle: string
 icon: JSX.Element
})
```

### Example usage

```ts
<Header icon={<Sword/>} desktopTitle="CHARACTER" mobileTitle="CHARACTER" endEhnacer={<button>Create new</button>}/>
```

## References

![Image](https://github.com/user-attachments/assets/191b84a1-5660-485b-8b89-4621696c7274)
![Image](https://github.com/user-attachments/assets/910d5e01-2d9d-4df6-ac75-18fd1c079f79)

## Resources

- Figma View: https://www.figma.com/design/csb1dd9yjNAg4sWTmVEZww/Arena-Mayhem?node-id=770-11931&t=htvhpzMvSyVVJfRj-0

---

> [!TIP]
> You can take Uber's Base Web Design System as an example: https://baseweb.design/components/button

**Comment by cyanidev** [comment]
![image](https://github.com/user-attachments/assets/65f3a406-bf59-4a3f-970d-35374ddc35d0)
![image](https://github.com/user-attachments/assets/0ebb970c-a5b8-48ec-9eeb-776297760b3e)
